pkg/retry/errors.go

```diff
@@ -35,3 +36,15 @@ func IsRetryableError(err error) bool {
 	}
 	return false
 }
+
+// IsConnectionError tells whether this error should reconnect to Database
+func IsConnectionError(err error) bool {
```

Comment:
should we always call `IsConnectionError` before `IsRetryableError`?

Reply:
for syncer.execute, we call `IsConnectionError` before `IsRetryableError`

Reply:
can we call `IsConnectionError` for more cases?

Reply:
ErrBadConn happen in Syncer is more common,it's difficult to see in loader after we use one db connection for one worker

Reply:
Why is it difficult to see in loader after we use one db connection for one worker.

Reply:
because loader has not set readTimeout, and it only execute simple DDL and DML.

Reply:
🤔  @amyangfei what's your opinion?

Reply:
> because loader has not set readTimeout, and it only execute simple DDL and DML.

we also set `readTimeout` now?

@3pointer @amyangfei ping

Reply:
seems this check is added in loader db operation

Reply:
But `ReadTimeout` is not set in loader?

Reply:
> seems this check is added in loader db operation

YES.

should we need to give a better name rather than `IsConnectionError`?